odoo/addons/base/models/ir_translation.py

```diff
@@ -193,7 +198,7 @@ def _auto_init(self):
         tools.create_unique_index(self._cr, 'ir_translation_unique', self._table,
                                   ['type', 'name', 'lang', 'res_id', 'md5(src)'])
         if not tools.index_exists(self._cr, 'ir_translation_code_unique'):
-            self._cr.execute("CREATE UNIQUE INDEX ir_translation_code_unique ON ir_translation (type, lang, md5(src)) WHERE type = 'code'")
+            self._cr.execute("CREATE UNIQUE INDEX ir_translation_code_unique ON ir_translation (type, lang, md5(src), is_web) WHERE type = 'code'")
```

Comment:
If this were to happen, then a pre-migration script should drop the `ir_translation_code_unique` index on the next update.